compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2106,12 +2106,12 @@ void handleDeclaredVarInForeach(BLangVariable variable, BType rhsType, SymbolEnv
                 break;
             case ERROR_VARIABLE:
                 BLangErrorVariable errorVariable = (BLangErrorVariable) variable;
+                errorVariable.setBType(rhsType);
```

Comment:
Is this correct when the RHS type isn't an error type?

Reply:
We won't go into this code block if RHS is not assignable to type of the variable node, which is a `BErrorType` in this case.